lib/Notification/FederationChatNotifier.php

```diff
@@ -0,0 +1,140 @@
+				$this->notificationManager->notify($notification);
+			} elseif ($this->isMentioned($participant, $message)) {
+				$notification = $this->createNotification($room, $message, 'mention');
+				$notification->setUser($participant->getAttendee()->getActorId());
+				$this->notificationManager->notify($notification);
+			} elseif ($this->isMentionedAll($room, $message)) {
+				$notification = $this->createNotification($room, $message, 'mention_all');
+				$notification->setUser($participant->getAttendee()->getActorId());
+				$this->notificationManager->notify($notification);
+			}
+		} elseif ($participant->getAttendee()->getNotificationLevel() === Participant::NOTIFY_ALWAYS
+			|| ($defaultLevel === Participant::NOTIFY_ALWAYS && $participant->getAttendee()->getNotificationLevel() === Participant::NOTIFY_DEFAULT)) {
+			$notification = $this->createNotification($room, $message, 'chat');
+			$notification->setUser($participant->getAttendee()->getActorId());
+			$this->notificationManager->notify($notification);
+		}
+	}
+
+	protected function isRepliedTo(Room $room, Participant $participant, array $metaData): bool {
+		if ($metaData[ProxyCacheMessage::METADATA_REPLYTO_TYPE] !== Attendee::ACTOR_FEDERATED_USERS) {
+			return false;
+		}
+
+		$repliedTo = $this->userConverter->convertTypeAndId($room, $metaData[ProxyCacheMessage::METADATA_REPLYTO_TYPE], $metaData[ProxyCacheMessage::METADATA_REPLYTO_ID]);
+		return $repliedTo['type'] === $participant->getAttendee()->getActorType()
+			&& $repliedTo['id'] === $participant->getAttendee()->getActorId();
+	}
+
+	protected function isMentioned(Participant $participant, ProxyCacheMessage $message): bool {
+		if ($participant->getAttendee()->getActorType() !== Attendee::ACTOR_USERS) {
+			return false;
+		}
+
+		foreach ($message->getParsedMessageParameters() as $parameter) {
+			if ($parameter['type'] === 'user' // RichObjectDefinition, not Attendee::ACTOR_USERS
+				&& $parameter['id'] === $participant->getAttendee()->getActorId()
+				&& empty($parameter['server'])) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	protected function isMentionedAll(Room $room, ProxyCacheMessage $message): bool {
+		foreach ($message->getParsedMessageParameters() as $parameter) {
+			if ($parameter['type'] === 'call' // RichObjectDefinition
+				&& $parameter['id'] === $room->getRemoteToken()) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Creates a notification for the given proxy message and mentioned users
+	 */
+	protected function createNotification(Room $chat, ProxyCacheMessage $message, string $subject, array $subjectData = []): INotification {
+		$subjectData['userType'] = $message->getActorType();
+		$subjectData['userId'] = $message->getActorId();
+
+		$notification = $this->notificationManager->createNotification();
+		$notification
+			->setApp('spreed')
+			->setObject('chat', $chat->getToken())
+			->setSubject($subject, $subjectData)
+			->setMessage($message->getMessageType(), [
```

Comment:
Display name for conversation seems incorrect
![image](https://github.com/nextcloud/spreed/assets/93392545/8333a362-7b2c-4c92-969c-db69f136cd63)
![image](https://github.com/nextcloud/spreed/assets/93392545/99e29c91-adf9-448d-9a94-b6e9cbe6eb3b)
![image](https://github.com/nextcloud/spreed/assets/93392545/545956f6-28f7-4655-9087-e2dc3700f480)

Reply:
Fixed (you need to send a new message to try it)

Reply:
Tested, perfect now